src/app/models/app-info.model.ts

```diff
@@ -31,24 +31,35 @@ export class AppInfo {
     public queueName: string,
     public submissionTime: number,
     public finishedTime: null | number,
+    public lastStateChangeTime: null | number,
     public applicationState: string,
     public allocations: AllocationInfo[] | null
-  ) {}
+  ) { }
 
   get formattedSubmissionTime() {
-    const millisecs = Math.round(this.submissionTime / (1000 * 1000));
-    return moment(millisecs).format('YYYY/MM/DD HH:mm:ss');
+    return this.formatTime(this.submissionTime)
   }
 
   get formattedFinishedTime() {
-    if (this.finishedTime) {
-      const millisecs = Math.round(this.finishedTime / (1000 * 1000));
-      return moment(millisecs).format('YYYY/MM/DD HH:mm:ss');
-    }
+    return this.formatTimeIfNotNull(this.finishedTime)
+  }
+
+  get formattedLastStateChangeTime() {
+    return this.formatTimeIfNotNull(this.lastStateChangeTime)
+  }
 
+  formatTimeIfNotNull(unformattedtime: null | number) {
+    if (unformattedtime) {
+      return this.formatTime(unformattedtime)
```

Comment:
Use code from formatTime here. No need of formatTime function.

Reply:
updated.